flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/data/FlinkAvroReader.java

```diff
@@ -63,6 +65,13 @@ public RowData read(RowData reuse, Decoder decoder) throws IOException {
     return DecoderResolver.resolveAndRead(decoder, readSchema, fileSchema, reader, reuse);
   }
 
+  @Override
+  public void setRowPositionSupplier(Supplier<Long> posSupplier) {
```

Comment:
I think it's a critical bug which we've missed to set the `posSupplier` to the `reader` since https://github.com/apache/iceberg/pull/1222.  But I'm still not sure why this [TestFlinkInputFormatReaderDeletes](https://github.com/apache/iceberg/blob/1d8164383baf0e343ade13b21ed0e4d4d02c56cd/flink/v1.13/flink/src/test/java/org/apache/iceberg/flink/source/TestFlinkInputFormatReaderDeletes.java)  did not cover the avro case.

Let me dig it into deeper and try to provide unit tests in the `TestFlinkInputFormatReaderDeletes`  to address this thing.

Reply:
Others from Chinese Iceberg group also encountered this bug,  they are stacktraces are the following:

```
Caused by: java.lang.NullPointerException
  at org.apache.iceberg.data.DeleteFilter.pos(DeleteFilter.java:108) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.deletes.Deletes$PositionSetDeleteFilter.shouldKeep(Deletes.java:157) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.util.Filter$Iterator.shouldKeep(Filter.java:49) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.io.FilterIterator.advance(FilterIterator.java:67) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.io.FilterIterator.hasNext(FilterIterator.java:50) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.io.FilterIterator.advance(FilterIterator.java:65) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.io.FilterIterator.hasNext(FilterIterator.java:50) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.flink.source.DataIterator.updateCurrentIterator(DataIterator.java:100) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.flink.source.DataIterator.hasNext(DataIterator.java:84) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.iceberg.flink.source.FlinkInputFormat.reachedEnd(FlinkInputFormat.java:104) ~[iceberg-flink-runtime-0.12.1-20211109.jar:?]
  at org.apache.flink.streaming.api.functions.source.InputFormatSourceFunction.run(InputFormatSourceFunction.java:89) ~[flink-dist_2.11-1.12.1.jar:1.12.1]
  at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:110) ~[flink-dist_2.11-1.12.1.jar:1.12.1]
  at org.apache.flink.streaming.api.operators.StreamSource.run(StreamSource.java:66) ~[flink-dist_2.11-1.12.1.jar:1.12.1]
  at org.apache.flink.streaming.runtime.tasks.SourceStreamTask$LegacySourceFunctionThread.run(SourceStreamTask.java:241) ~[flink-dist_2.11-1.12.1.jar:1.12.1]
```

Reply:
Should we separate this fix into other issue? This doesn't seem to be related to the unittest.

Reply:
Yes,  that's exactly what I plan to do.  Thanks @Reo-LEI .

Reply:
Created separate PR for this: https://github.com/apache/iceberg/pull/3540

Reply:
Merged #3540.

Reply:
As the separate PR #3550 has been merged,  Let's update this PR.